src/Glpi/Application/Environment.php

```diff
@@ -0,0 +1,256 @@
+                    implode('`, `', $allowed_keys)
+                )
+            );
+        }
+    }
+
+    /**
+     * See SystemConfigurator::computeConstants() for all available values that
+     * can be overridden.
+     */
+    public function getConstantsOverride(string $root_dir): array
+    {
+        return match ($this) {
+            default => [],
+            self::TESTING     => [
+                'GLPI_CONFIG_DIR'               => $root_dir . '/tests/config',
+                'GLPI_VAR_DIR'                  => $root_dir . '/tests/files',
+                'GLPI_STRICT_ENV'               => true,
+                'GLPI_SERVERSIDE_URL_ALLOWLIST' => [
+                    // default allowlist entries
+                    '#^http://[^@:]+(:80)?(/.*)?$#',
+                    '#^https://[^@:]+(:443)?(/.*)?$#',
+                    '#^feed://[^@:]+(/.*)?$#',
+
+                    // calendar mockups
+                    '/^file:\/\/.*\.ics$/',
+                ],
+                'PLUGINS_DIRECTORIES'           => [
+                    $root_dir . '/plugins',
+                    $root_dir . '/tests/fixtures/plugins',
+                ],
+            ],
+            self::DEVELOPMENT => [
+                'GLPI_WEBHOOK_ALLOW_RESPONSE_SAVING' => '1'
+            ],
+        };
+    }
+
+    public function getLogLevel(): string
+    {
+        return match ($this) {
+            default           => LogLevel::WARNING,
+            self::TESTING     => LogLevel::DEBUG,
+            self::DEVELOPMENT => LogLevel::DEBUG,
+        };
+    }
+
+    /**
+     * Should debug, info and notice be reported ?
+     * See ErrorHandler::configureErrorReporting().
+     *
+     * Note: this seems weird and should not necessary as we already have
+     * getLogLevel() to define this behavior in the AbstractLogger class.
+     *
+     * TODO: investigate and remove this method and its usage if needed.
+     */
+    public function shouldReportLowLevelErrors(): bool
+    {
+        return match ($this) {
+            default           => false,
+            self::TESTING     => true,
+            self::DEVELOPMENT => true,
+        };
+    }
+
+    /**
+     * Will the files of this environment change ?
+     */
+    public function shouldWatchForRessourcesChanges(): bool
+    {
+        // Only production/staging environment are considered as environments
+        // where resources are not supposed to change.
+        // In others environments, we must match for changes.
+        return match ($this) {
+            default           => false,
+            self::DEVELOPMENT => true,
+            // TODO: when running on the CI, TESTING should return false as the
+            // code won't change. Should be easy to set up.
+            // Or maybe it should always be false and the test bootstrap file
+            // should clear the caches on start up, guaranteeing a clean env ?
+            self::TESTING     => true,
```

Comment:
Clearing the cache during the bootstrap is, IMHO, the only solution. Indeed, if you run the CI env on your local machine (with the `tests/run_tests.sh` script), some cached files may be outdated and your tests suite may fail due to this.

The cache cleaning should probably be done with a loop on a `RecursiveDirectoryIterator` and should exit with a failure if it cannot remove one of the cached files.

Reply:
This change was enough in the bootstrap on my end:

```php
$cache_manager = new CacheManager();
$cache_manager->resetAllCaches();
```

I will not commit it because we also need to do it for the e2e and there is no boostrap file for cypress (there is one in the playwright PR tho).